Purchases/Attribution/AttributionFetcher.swift

```diff
@@ -138,4 +134,56 @@ private extension AttributionFetcher {
         }
     }
 
+    @available(iOS 14.0.0, tvOS 14.0.0, *)
+    private var fetchAuthorizationStatus: FakeTrackingManagerAuthorizationStatus {
+        let needsTrackingAuthorization = self.needsTrackingAuthorization
```

Comment:
Nit: this is only used inside the `guard` so you could move it there.

Reply:
It's only used in the `guard` but the return value can be different. If this guard fails, we return 
`!needsTrackingAuthorization`
If the next guard fails, we return 
`false`

I think I should leave it to preserve that logic since each guard is really a unique situation we're testing for.

Reply:
Merging for now, but if we want to revisit the logic, we can definitely do that, I just don't want to block 😄